src/pfe/file-watcher/server/test/functional-test/suites/project/tests/project-specification.ts

```diff
@@ -431,7 +506,9 @@ export function projectSpecificationTest(socket: SocketIO, projData: ProjectCrea
 
             const dockerfile = path.join(projData.location, "Dockerfile");
             const fileOutput = await fs.readFileSync(dockerfile, {encoding: "utf-8"});
-            await fs.writeFileSync(dockerfile, fileOutput.replace(new RegExp(`EXPOSE ${defaultInternalPort}`, "g"), `EXPOSE ${testExposedPort}`), {encoding: "utf-8"});
+            if (projData.projectType != "spring") {
```

Comment:
Cant we use your default one port exposed object in this if block condition rather than hardcoding spring

Reply:
I don't think so. This if statement is required for spring projects (and latter on for liberty) because neither of those project has `EXPOSE <PORT>` on their dockerfile. So we don't want to write to the dockerfile on those cases to update the port.

Reply:
Understood